packages/payment-experience-api/src/api/paytrailCardUpdateRedirectSuccessController.ts

```diff
@@ -0,0 +1,63 @@
+import { AbstractController, logger } from '@verkkokauppa/core'
+import type { Request, Response } from 'express'
+import { URL } from 'url'
+import { getOrderAdmin } from '@verkkokauppa/order-backend'
+import { checkPaytrailCardUpdateReturnUrl } from '@verkkokauppa/payment-backend'
+
+export class PaytrailCardUpdateRedirectSuccessController extends AbstractController {
+  protected readonly requestSchema = null
+
+  private static success = (url: URL, orderId: string | undefined) => {
+    url.pathname = `${orderId}/card-update-success`
+    return url
+  }
+
+  private static fault = (url: URL, orderId?: string) => {
+    url.pathname = orderId
+      ? `${orderId}/card-update-failed`
+      : 'card-update-failed'
```

Comment:
this else statement should have orderId too

Reply:
I checked and this was ok, frontend redirects to error page no action needed for now.

Reply:
It's only for the case where orderId is missing from the path for one reason or another. Though that should never happen anyway because express should return 404 without orderId in the path